include/libtorrent/span.hpp

```diff
@@ -105,12 +105,19 @@ namespace libtorrent
 			return { data() + offset, count };
 		}
 
-		T& operator[](std::size_t const idx)
+		T& operator[](std::size_t const idx) const
 		{
 			TORRENT_ASSERT(idx < m_len);
 			return m_ptr[idx];
 		}
 
+		T& operator[](int const idx) const
```

Comment:
looking at it, I believe I missed the `IndexType` in the return type of the methods.